### PR TITLE
fix check type of predicate and list comprehension

### PR DESCRIPTION
**Pull request by jievince**

### CODE REVIEW

File: src/common/expression/ListComprehensionExpression.cpp [hunk 1]
```diff
@@ -13,6 +13,13 @@ const Value& ListComprehensionExpression::eval(ExpressionContext& ctx) {
     List ret;
 
     auto& listVal = collection_->eval(ctx);
+    if (listVal.isNull()) {
```
Comment (CPWstatic) [inline src/common/expression/ListComprehensionExpression.cpp:16]: We could return the listVal directly.
```
if (listVal.isNull() || listVal.empty()) {
    result = listVal;
    return result_;
}
```

Comment (jievince) [reply src/common/expression/ListComprehensionExpression.cpp:16]: > We could return the listVal directly.
> 
> ```
> if (listVal.isNull() || listVal.empty()) {
>     result = listVal;
>     return result_;
> }
> ```

ok()

**Review by yixinglu — DISMISSED** [review]

**Review by CPWstatic — APPROVED** [review]

**Comment by yixinglu** [comment]
please solve the conflicts.

**Comment by jievince** [comment]
> please solve the conflicts.

This PR is unuseful now. Because the content of this PR has been put into the pr of `Reduce Expression` which has been merged.

**Pull request closed without merging**